convert2rhel/pkghandler.py

```diff
@@ -313,11 +314,25 @@ def print_pkg_info(pkgs):
     return pkg_table
 
 
-def get_pkg_nvra(pkg_obj):
-    return "%s-%s-%s.%s" % (pkg_obj.name,
-                            pkg_obj.version,
-                            pkg_obj.release,
-                            pkg_obj.arch)
+def get_pkg_nevra(pkg_obj):
+    """Get package NEVRA as a string: name, epoch, version, release, architecture.
+
+    Epoch is included only when non-zero. However it's on a different position when printed by YUM or DNF:
+      YUM - epoch before name: "7:oraclelinux-release-7.9-1.0.9.el7.x86_64"
+      DNF - epoch before version: "oraclelinux-release-8:8.2-1.0.8.el8.x86_64"
+    """
+    if pkgmanager.TYPE == 'yum':
+        return "%s%s-%s-%s.%s" % ("" if pkg_obj.epoch == 0 else str(pkg_obj.epoch) + ":",
+                                  pkg_obj.name,
+                                  pkg_obj.version,
+                                  pkg_obj.release,
+                                  pkg_obj.arch)
+    # DNF
+    return "%s-%s%s-%s.%s" % (pkg_obj.name,
+                              "" if pkg_obj.epoch == 0 else str(pkg_obj.epoch) + ":",
+                              pkg_obj.version,
+                              pkg_obj.release,
+                              pkg_obj.arch)
```

Comment:
Should our checks not be DNF by default?
```suggestion
    if pkgmanager.TYPE == 'dnf':
        return "%s-%s%s-%s.%s" % (pkg_obj.name,
                                  "" if pkg_obj.epoch == 0 else str(pkg_obj.epoch) + ":",
                                  pkg_obj.version,
                                  pkg_obj.release,
                                  pkg_obj.arch)
    # YUM
    return "%s%s-%s-%s.%s" % ("" if pkg_obj.epoch == 0 else str(pkg_obj.epoch) + ":",
                              pkg_obj.name,
                              pkg_obj.version,
                              pkg_obj.release,
                              pkg_obj.arch)
```

Reply:
From my point of view it doesn't matter what's the default. Why do you think the other way is better?

Reply:
Thinking about it, probably better to have a check on yum explicitly. But having DNF as default is something we should do to make it easier to remove yum once we no longer support systems that uses yum

Reply:
https://github.com/oamg/convert2rhel/pull/112/commits/92ee17e0ecadcc60f43ffe03347aa02a623f75f9